base/src/com/google/idea/blaze/base/bazel/LocalInvoker.java

```
@@ -0,0 +1,275 @@
+ */
+public class LocalInvoker extends AbstractBuildInvoker1 {
+  private static Logger logger = Logger.getInstance(LocalInvoker.class);
+  private final BuildBinaryType buildBinaryType;
+
+  public LocalInvoker(Project project, BlazeContext blazeContext, BuildSystem buildSystem, BuildBinaryType binaryType) {
+    super(project, blazeContext, buildSystem, binaryPath(binaryType, project));
+    this.buildBinaryType = binaryType;
+  }
+
+  @Override
+  public BuildResultHelper createBuildResultHelper() {
+    return new BuildResultHelperBep();
+  }
+
+  @Override
+  public BlazeCommandRunner getCommandRunner() {
+    return new CommandLineBlazeCommandRunner();
+  }
+
+  @Override
+  public boolean supportsHomeBlazerc() {
+    return true;
+  }
+
+  @Override
+  public boolean supportsParallelism() {
+    return false;
+  }
+
+  @Override
+  public ImmutableSet<Capability> getCapabilities() {
+    return ImmutableSet.of(Capability.IS_LOCAL, Capability.SUPPORTS_CLI);
+  }
+
+  @Override
+  public BuildEventStreamProvider invoke(BlazeCommand.Builder blazeCommandBuilder)
+    throws BuildException {
+    try {
+      performGuardCheck(project, blazeContext);
+    }
+    catch (ExecutionDeniedException e) {
+      throw new BuildException(e.getMessage(), e);
+    }
+    File outputFile = BuildEventProtocolUtils.createTempOutputFile();
+    BuildResult buildResult =
+      issueBuild(blazeCommandBuilder, WorkspaceRoot.fromProject(project), blazeContext, outputFile);
+    if (blazeCommandBuilder.build().getName().equals(BlazeCommandName.BUILD)) {
+      BuildDepsStatsScope.fromContext(blazeContext)
+        .ifPresent(stats -> stats.setBazelExitCode(buildResult.exitCode));
+    }
+    return getBepStream(outputFile);
+  }
+
+  @Override
+  public InputStream invokeQuery(BlazeCommand.Builder blazeCommandBuilder) {
+    try {
+      performGuardCheck(project, blazeContext);
+    }
+    catch (ExecutionDeniedException e) {
+      logger.error(e);
```

This looks like error swallowing to me.

---

Yes, not perfect but `logger.error` should be fine. Also, not worth deviating from AOSP for this in my opinion.